Refactoring maven packages

d-michail opened the pull request:
Refactored the maven packages to have a separate "bundle" module which builds one single "uber" jar. I excluded this bundle and the older dist module from the deploy plugin.

----

- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/Become-a-Contributor>
- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/How-to-make-your-first-%28code%29-contribution>
- [x] I added [unit tests](https://github.com/jgrapht/jgrapht/wiki/Unit-testing)
- [x] I added [documentation](https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation)
- [x] I followed the [Coding and Style Conventions](https://github.com/jgrapht/jgrapht/wiki/Coding-and-Style-Conventions)
- [x] I **have not** modified `HISTORY.md` or `CONTRIBUTORS.md`
- [x] I ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)

jsichi commented:
Sweet!  I tested by editing the classpath in the demo jar to reference the bundle jar instead of the individual jars, and all was well.

There are a few remaining references to uber jars:

* in README.md
* in jgrapht-touchgraph/pom.xml

And there's one place in the wiki that we'll need to update as part of release:

https://github.com/jgrapht/jgrapht/wiki/How-to-use-JGraphT-as-a-dependency-in-your-projects

d-michail commented:
I cleaned up the remaining parts for the uber jars. I also added some imports in touchgraph (I do not think that anyone is actually using this, as it was completely broken).

jsichi commented:
Thanks!
